Sources/NIOHTTP1/NIOTypedHTTPServerUpgraderStateMachine.swift

```diff
@@ -382,5 +382,23 @@ struct NIOTypedHTTPServerUpgraderStateMachine<UpgradeResult> {
         }
     }
 
+    @usableFromInline
+    enum CloseInboundAction {
+        case close
+        case `continue`
+    }
+
+    @inlinable
+    mutating func closeInbound() -> CloseInboundAction {
+        switch self.state {
+        case .initial, .awaitingUpgrader:
+            self.state = .finished
+            return .close
+
+        default:
```

Comment:
Can we please spell out all cases explicitly so if we ever add one we get a compiler error here